engine/new_runtime/src/buffer.rs

```diff
@@ -0,0 +1,370 @@
+use crate::boxed::Boxed;
+use crate::locked_memory::{*, MemoryConfiguration::*, MemoryError::*};
+use crate::types::{Bytes, Zeroed, Randomized, ConstEq};
+use core::fmt::{self, Debug, Formatter};
+use core::ops::{Deref, DerefMut};
+use core::marker::PhantomData;
+
+use serde::{
+    de::{Deserialize, Deserializer, SeqAccess, Visitor},
+    ser::{Serialize, SerializeSeq, Serializer},
+};
+
+/// GuardedMemory is used when we want to store sensitive non encrypted data
+/// This shall always be short lived
+#[derive(Clone, Eq)]
+pub struct Buffer<T: Bytes> {
+    boxed : Boxed<T>, // the boxed type of current GuardedVec
+}
+
+pub struct Ref<'a, T: Bytes> {
```

Comment:
nitpicking: wouldn't the `Ref` struct conflict with `core::cell::Ref`? Is there an advantage not implement `Deref` and `DerefMut`, but have a distinct type, expect for the lifetime?